server/src/test/kotlin/fi/oph/ludos/assignment/AssignmentFiltersTest.kt

```diff
@@ -0,0 +1,272 @@
+
+        val assignmentsOut = objectMapper.readValue(assignments, Array<TestPuhviOut>::class.java)
+
+        val actualNumbersInName = assignmentsOut.flatMap { assignment ->
+            Regex("\\d+").findAll(assignment.nameFi)
+                .map { it.value.toInt() }
+                .toList()
+        }
+
+        assertEquals(expectedNumbersInName.sorted(), actualNumbersInName.sorted())
+    }
+
+    private fun testLdFilterOptions(
+        aine: String?, lukuvuosi: String?, expectedNumbersInName: List<Int>
+    ) {
+        val ldFilters = LdAssignmentFilter(
+            orderDirection = "desc", aine = aine, lukuvuosi = lukuvuosi
+        )
+
+        val assignments = mockMvc.perform(getLdAssignments(Exam.LD, ldFilters)).andExpect(
+            MockMvcResultMatchers.status().isOk()
+        ).andReturn().response.contentAsString
+
+        val assignmentsOut = objectMapper.readValue(assignments, Array<TestLdOut>::class.java)
+
+        val actualNumbersInName = assignmentsOut.flatMap { assignment ->
+            Regex("\\d+").findAll(assignment.nameFi)
+                .map { it.value.toInt() }
+                .toList()
+        }
+
+        assertEquals(expectedNumbersInName.sorted(), actualNumbersInName.sorted())
+    }
+
+    @Test
+    fun testFiltersWithNonAllowedValues() {
+        // suko
+        testNonAllowedFilterOptions(
+            Exam.SUKO,
+            "?orderDirection=desc&oppimaara=!&tehtavatyyppisuko=001",
+            "oppimaara: must match \"^[a-zA-Z0-9,]+\$\""
+        )
+        testNonAllowedFilterOptions(
+            Exam.SUKO, "?orderDirection=desc&tehtavatyyppisuko=a", "tehtavatyyppisuko: must match \"^[0-9,]+\$\""
+        )
+        testNonAllowedFilterOptions(
+            Exam.SUKO,
+            "?oppimaara=TKFIA1,TKRUAI,TKFIB3&tavoitetaitotaso=sdas,0008,0009&orderDirection=desc",
+            "tavoitetaitotaso: must match \"^[0-9,]+\$\""
+        )
+        testNonAllowedFilterOptions(Exam.SUKO, "?aihe=asd&orderDirection=desc", "aihe: must match \"^[0-9,]+\$\"")
+        testNonAllowedFilterOptions(
+            Exam.SUKO, "?aihe=015,002,00A&orderDirection=desc", "aihe: must match \"^[0-9,]+\$\""
+        )
+        // puhvi
+        testNonAllowedFilterOptions(
+            Exam.PUHVI, "?orderDirection=desc&lukuvuosi=20212022,20202021A", "lukuvuosi: must match \"^[0-9,]+\$\""
+        )
+        testNonAllowedFilterOptions(
+            Exam.PUHVI, "?tehtavatyyppipuhvi=aASs", "tehtavatyyppipuhvi: must match \"^[0-9,]+\$\""
+        )
+        // ld
+        testNonAllowedFilterOptions(Exam.LD, "?aine=aASs", "aine: must match \"^[0-9,]+\$\"")
+        testNonAllowedFilterOptions(Exam.LD, "?aine=", "aine: must match \"^[0-9,]+\$\"")
+    }
+
+    private fun testNonAllowedFilterOptions(exam: Exam, filterStr: String, expectedErrorString: String) {
+        val errorStr =
+            mockMvc.perform(getAssignmentsWithAnyFilter(exam, filterStr)).andReturn().response.contentAsString
```

Comment:
Tsekkaisin HTTP-koodinkin:

```suggestion
        val errorStr =
            mockMvc.perform(getAssignmentsWithAnyFilter(exam, filterStr))
                .andExpect(MockMvcResultMatchers.status().isBadRequest)
                .andReturn().response.contentAsString
```